### PR TITLE
Bug 1210720 - Migrate Music to l20n/service & l20n/client

### PR DESCRIPTION
**Pull request by stasm**
https://bugzilla.mozilla.org/show_bug.cgi?id=1210720

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=0d0a413086ff764973e5fd01a0ba90d8cd5e99a4

File: shared/js/l20n/client.js [hunk 1]
```diff
@@ -0,0 +1,432 @@
...
+      return this.service.broadcast(type, data);
+    }
+
+    return { Client, Service, channel, broadcast };
+  });
+  modules.set('runtime/bridge/index-client', function () {
+    const { Client, channel } = getModule('runtime/bridge/bridge');
+    const { View } = getModule('bindings/html/view');
+
+    const client = new Client('l20n', channel);
```
Comment (justindarc) [inline shared/js/l20n/client.js]:
```
const client = new Client({
  service: 'l20n',
  endpoint: channel,
  timeout: false
});
```

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=cc01e80e38330d9c132f7981af069c80bfbb3440

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=1e467143a35016fdb62ad6c36d54327df62a6105

File: apps/music/index.html [hunk 2]
```diff
@@ -18,7 +18,8 @@
   <link rel="stylesheet" href="/css/app.css">
   <!-- Deferred Scripts -->
   <script defer src="/components/bridge/bridge.js"></script>
-  <script defer src="/shared/js/intl/l20n.js" type="application/javascript;version=1.8"></script>
+  <script defer src="/shared/js/intl/l20n-service.js" type="application/javascript;version=1.8"></script>
+  <script defer src="/shared/js/intl/l20n-client.js" type="application/javascript;version=1.8"></script>
```
Comment (justindarc) [inline apps/music/index.html]: Is there any common code between these? At least in the case of bridge.js, there is shared code between client.js and service.js which is why they put a full bridge.js in there for cases where you need both to reduce the overhead. Just wondering.

Comment (stasm) [inline apps/music/index.html]: There very little common code.  In fact, it's just these 25 lines: https://github.com/l20n/l20n.js/blob/962b9a4b82b91a25de63472b2fb783d34355707b/src/bindings/html/shims.js. I think I would prefer to keep the fies separate, at least until we need more shared code;  it makes it easier to debug l20n and see which file the errors are coming from.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=32433bfa612e5efa82754ea43102066ec0decb22

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=81918af334ce5cf9d0b83f9f17140914d83b5862

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3aa35e4cdce394c1bc0f8a4191ccff0fa6ad4104

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=da5d00792ac063518e20824110b2fbf7f8a63d7e

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=f15515f15993684c55a8cf0568b08a3c409dac85

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=36556e599eca1bd91aaf45b8ceb1b020fa74b72d

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=62a6dd4f3ab03b3e90152c35e6270519b600a1dd

**Pull request merged**